src/gluonnlp/vocab/vocab.py

```diff
@@ -434,16 +507,26 @@ def from_json(cls, json_str):
         Vocab
         """
         vocab_dict = json.loads(json_str)
-
+        token_to_idx = vocab_dict.get('token_to_idx')
         unknown_token = vocab_dict.get('unknown_token')
-        vocab = cls(unknown_token=unknown_token)
-        vocab._idx_to_token = vocab_dict.get('idx_to_token')
-        vocab._token_to_idx = vocab_dict.get('token_to_idx')
-        if unknown_token:
-            vocab._token_to_idx = DefaultLookupDict(vocab._token_to_idx[unknown_token],
-                                                    vocab._token_to_idx)
-        vocab._reserved_tokens = vocab_dict.get('reserved_tokens')
-        vocab._padding_token = vocab_dict.get('padding_token')
-        vocab._bos_token = vocab_dict.get('bos_token')
-        vocab._eos_token = vocab_dict.get('eos_token')
+        padding_token = vocab_dict.get('padding_token')
+        bos_token = vocab_dict.get('bos_token')
+        eos_token = vocab_dict.get('eos_token')
+        special_tokens = [unknown_token, padding_token, bos_token, eos_token]
+        reserved_tokens = vocab_dict.get('reserved_tokens')
+
+        # workaround reserved and special tokens being serialized together
+        reserved_tokens = [
+            t for t in reserved_tokens if t not in special_tokens
+        ]
+
+        vocab = cls(
+            counter=count_tokens(token_to_idx.keys()),
+            unknown_token=unknown_token,
+            padding_token=padding_token,
+            bos_token=bos_token,
+            eos_token=eos_token,
+            reserved_tokens=reserved_tokens,
+            token_to_idx=token_to_idx,
+            identifiers_to_tokens=vocab_dict.get('identifiers_to_tokens'))
```

Comment:
This might not exist in previous serialized vocab files. Provide a default value?

Reply:
`dict.get` returns None if the value does not exist. Ie. the default value is None